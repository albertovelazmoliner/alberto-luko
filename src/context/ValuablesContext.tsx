import React, {
  createContext,
  Dispatch,
  SetStateAction,
  useState,
} from "react";
import { Valuable } from "../domain";

type IValuablesContext = {
  valuables: Valuable[];
  setValuables: Dispatch<SetStateAction<Valuable[]>>;
}

const ValuablesContext = createContext<IValuablesContext>({} as IValuablesContext);

export function ValuablesContextProvider(props: { children: React.ReactNode }) {
  const [valuables, setValuables] = useState<Valuable[]>([]);

  return (
    <ValuablesContext.Provider value={{ valuables, setValuables }}>
      {props.children}
    </ValuablesContext.Provider>
  );
}

export default ValuablesContext;
