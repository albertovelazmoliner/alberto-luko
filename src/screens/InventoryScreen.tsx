import { useContext } from "react";
import { FlatList, StyleSheet, View } from "react-native";
import { Title } from "../components/Title";
import { ValuableCard } from "../components/valuableCard";
import ValuableContext from "../context/ValuablesContext";
import { RootTabScreenProps } from "../navigation/types";
import { colors } from "../theme/colors";

export default function InventoryScreen({
  navigation,
  route,
}: RootTabScreenProps<"Inventory">) {
  const handleAddButtonPress = () => navigation.navigate("AddItem");

  const { valuables } = useContext(ValuableContext);

  return (
    <View style={styles.container}>
      <Title onButtonPress={handleAddButtonPress}>{route.name}</Title>
      <FlatList
        data={valuables}
        renderItem={({ item }) => <ValuableCard item={item} />}
        numColumns={2}
        keyExtractor={(item) => `${item.id}`}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 20,
    backgroundColor: colors.background,
  },
});
