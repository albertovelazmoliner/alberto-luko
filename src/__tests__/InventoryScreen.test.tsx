import InventoryScreen from "../screens/InventoryScreen";
import renderer from "react-test-renderer";
import React from "react";
import { ValuablesContextProvider } from "../context/ValuablesContext";

const navigation: any = { navigate: jest.fn() };
const route: any = { name: "Inventory" };

const whenStable = async () => {
  await renderer.act(async () => {
    await new Promise((resolve) => {
      setTimeout(resolve, 0);
    });
  });
};

describe("test Inventory Screen is rendered", () => {
  it("foo", async () => {
    const screen = renderer.create(
      <ValuablesContextProvider>
        <InventoryScreen navigation={navigation} route={route} />
      </ValuablesContextProvider>
    );

    expect(screen.toJSON()).toMatchSnapshot();

    await whenStable();
  });
});
