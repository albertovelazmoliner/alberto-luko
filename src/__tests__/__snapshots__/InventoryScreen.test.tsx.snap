// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Inventory Screen is rendered foo 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "height": 42,
        "justifyContent": "space-between",
        "marginTop": 99,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Circular-Bold",
          "fontSize": 34,
          "lineHeight": 42,
        }
      }
    >
      Inventory
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "opacity": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#2D50E6",
            "borderRadius": 14,
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
